### ISSUE TITLE
undefined reference to `clock_gettime' in eidos_functions.cpp

### ISSUE THREAD

**Issue by xin-huang** [MesserLab/SLiM#63]
Hi I am trying to complie SLiM-v3.3.1 in our linux server, but it returns errors:
```
Linking CXX executable slim
CMakeFiles/slim.dir/eidos/eidos_functions.cpp.o: In function `Eidos_ExecuteFunction_clock(Eidos_intrusive_ptr<EidosValue> const*, int, EidosInterpreter&)':
eidos_functions.cpp:(.text+0xb36a): undefined reference to `clock_gettime'
CMakeFiles/slim.dir/eidos/eidos_functions.cpp.o: In function `Eidos_ExecuteLambdaInternal(Eidos_intrusive_ptr<EidosValue> const*, EidosInterpreter&, bool)':
eidos_functions.cpp:(.text+0x268cb): undefined reference to `clock_gettime'
eidos_functions.cpp:(.text+0x26946): undefined reference to `clock_gettime'
collect2: error: ld returned 1 exit status
make[3]: *** [slim] Error 1
make[2]: *** [CMakeFiles/slim.dir/all] Error 2
make[1]: *** [CMakeFiles/slim.dir/rule] Error 2
make: *** [slim] Error 2

```
But I can successfully compile SLiM-v3.3 after [fixing build error on CenTOS 7 under gcc6.1.0](https://github.com/MesserLab/SLiM/commit/2f3c55e34b72c8ae3780f49e6f787179a12d5ab0).

**Comment by bhaller** [comment]
I have just committed the addition of a header include that should fix the problem.  Different OSes have different header requirements, unfortunately, so these things can slip through the cracks.  You should be able to build the GitHub head of the master branch now (let me know if not).  Thanks for the report.

**Comment by xin-huang** [comment]
Hi Ben, 

I have tried the version in the master branch, but still got the same problem.

**Comment by bhaller** [comment]
Really?  The exact same errors?  That doesn't make sense to me; the errors were "undefined reference to `clock_gettime'", clock_gettime() is defined in time.h, I added an include for time.h, the problem ought to be fixed.  Can you double-check that the error is the same?  And if it is, can you try to find out where clock_gettime() is defined on your system?  Sorry, I'm not able to figure this out on my end easily, since I don't have a CenTOS machine, etc.  If you are able to do a little detective work, that would be helpful.

**Comment by bhaller** [comment]
My build testing on travis-ci was broken for a while, but I've just gotten it working again and it looks like the Linux build there of SLiM is presently passing.  I'm going to close this issue now since you haven't replied to my comment above, but please feel free to reopen it if you have further issues with your build of the master branch.  Thanks for the report.

**Comment by xin-huang** [comment]
Hi Ben,

I modified CMakeLists.txt with `target_link_libraries(${TARGET_NAME} rt)`.
Now it works.
[CMakeLists.txt](https://github.com/MesserLab/SLiM/files/3941565/CMakeLists.txt)

**Comment by bhaller** [comment]
Hi!  Can you tell me more about that?  clock_gettime() should be part of the standard library, as I understand it; on your system is it not in the standard library, such that you have to add a link to some external library?  If you know more about this, I'd like to hear.

**Comment by xin-huang** [comment]
Hi Ben, 

I think this is because the glibc version of our server is old.
From [the man page of clock_gettime](http://man7.org/linux/man-pages/man2/clock_gettime.2.html), it is necessary to link with `librt` before glibc 2.17

**Comment by bhaller** [comment]
> I think this is because the glibc version of our server is old.

Looks like glibc 2.17 was released in 2012, and nobody else has reported this problem, so I think I'll keep the bug closed.  Thanks for the info, it will prove useful if anybody else runs into this.  :->

**Comment by bhaller** [comment]
Fixed by #67 .
